ocular/src/chain/client/automated_tx_handler.rs

```diff
@@ -0,0 +1,292 @@
+#![warn(unused_qualifications)]
+#![allow(unused_imports)]
+// Clippy broken; doesn't recognize certain imports are used and sees them as unused
+
+use crate::{
+    chain::{
+        client::tx::{Account, TxMetadata},
+        config::ChainClientConfig,
+    },
+    error::{AutomatedTxHandlerError, TxError},
+    keyring::Keyring,
+};
+use bip32::PrivateKey;
+use cosmrs::{
+    crypto::{secp256k1::SigningKey, PublicKey},
+    rpc, AccountId, Coin,
+};
+use serde::{Deserialize, Serialize};
+use signatory::{pkcs8::der::Document, pkcs8::LineEnding};
+use std::{fs, path::Path, str::FromStr};
+use tendermint_rpc::endpoint::broadcast::tx_commit::Response;
+
+use super::ChainClient;
+
+// Toml structs
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct NonDelegatedToml<'a> {
+    pub sender: NonDelegatedSender<'a>,
+
+    #[serde(borrow)]
+    pub transactions: Vec<Transaction<'a>>,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct NonDelegatedSender<'a> {
+    pub prefix: &'a str,
+
+    pub source_private_key_path: &'a str,
+
+    pub chain_id: &'a str,
+}
```

Comment:
A couple of things:
1. `chain_id` is inside of the client already no?
2. This can be replaced by the yet-to-be-merged `Account` type I think.

Reply:
Yeah that's true, we might as well get it from the chain client. Re 2, i believe we still need the source private key path as that might not necessarily be loaded into the keystore (e.g. could be a custom key a user has lying around). We can always convert it to the new account type though.

Reply:
Hm, raises a design question. It's weird to me to have the go-to method for batch transactions require a path instead of going through the keyring or Accounts. Maybe we could have a separate method where a private key path could be passed in, or maybe we require that they load an `Account` from the key first before using? I could add an `Account::from(key_path);` method or some such in my branch.

Reply:
Hmm I think maybe we add in a new method to the keyring to pass in a private key path? That seems like a potential use case we might want to support regardless of this PR so we could kill 2 birds with 1 stone there.

Reply:
Added in https://github.com/PeggyJV/ocular/issues/21